Adds `SimpleCustomFillModel` to `CustomModelsAlgorithm`

AlexCatarino opened the pull request:
#### Description
The simple fill model shows how to implement a simpler version of the most popular order fills: Market, Stop Market, and Limit.

This model was tested on QuantConnect Cloud and will serve as an additional example since we don't have an example that does not reuse the base class method.

#### Related Issue
N/A

#### Motivation and Context
Documentation/Examples

#### Requires Documentation Change
No. We already point to `CustomModelsAlgorithm`.

#### How Has This Been Tested?
QuantConnect Cloud.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. <!--- If not applicable, please explain why -->
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

AlexCatarino committed:
Adds `SimpleCustomFillModel` to `CustomModelsAlgorithm`

The simple fill model shows how to implement a simpler version of the most popular order fills: Market, Stop Market and Limit.

This model was tested on QuantConnect Cloud, and will serve as additonal example, since we don't have an example that does not reuse the method of the base class.

Martin-Molinero commented on Algorithm.Python/CustomModelsAlgorithm.py:
```diff
@@ -117,3 +117,56 @@ def HasSufficientBuyingPowerForOrder(self, parameters):
+        if order.Status == OrderStatus.Canceled: return fill
+        limit_price = order.LimitPrice
+
+        trade_bar = asset.Cache.GetData[TradeBar]()
```
Suggest adding checks in case trade_bar is None/null but this won't work for tick resolution. Should we maybe just rely on `SecurityCache.BidPrice/AskPrice` for this simple version?

AlexCatarino committed:
Handles Tick Resolution Case

Tick-resolution data doesn't have TradeBar. We can use the security price, since it represents a trade (`TickTrade`).

Martin-Molinero reviewed (approved):
Nice! 👍
